pkg/ebpf/tracee.go

```diff
@@ -778,7 +778,11 @@ func (t *Tracee) populateBPFMaps() error {
 
 	cZero := uint32(0)
 	configVal := make([]byte, 208)
-	binary.LittleEndian.PutUint32(configVal[0:4], uint32(os.Getpid()))
+	selfPid := 0
```

Comment:
@rafaeldtinoco @yanivagman If I'm not mistaken this is the scheduler's PID right? so this might be a problem.

Reply:
I'm very curious to learn if that's true

Reply:
Based on this https://www.baeldung.com/linux/process-pid-0

Reply:
I believe that is not currently (modern kernels) accurate. If you follow x86_64 arch (at least) logic:

1. x86_64_start_kernel()
2. x86_64_start_reservations()
3. start_kernel()

at the very end of start_kernel() you will see:

![image](https://user-images.githubusercontent.com/7395852/196836362-d5847df0-1389-4319-80fb-ffad48dd0b4d.png)

Ending up with: rest_init() function.

If you read rest_init() logic, you will see that:

1. The initial task (PID=1) is created, and it will run kernel_init() when scheduled, which will call the init process:
```C
	pid = user_mode_thread(kernel_init, NULL, CLONE_FS);
```

2. Then a kernel thread, called kthreadadd, is created and that is a task that will always be in charge of creating more kernel threads, as needed, in an infinite loop (scheduled in and out), as you can see in the function "kthreadadd".
```C
	pid = kernel_thread(kthreadd, NULL, CLONE_FS | CLONE_FILES);
```

3. The "kernel_init" function waits for a completion at the beginning, that only happens after "kthreadadd" task exists:
```C
	complete(&kthreadd_done);
```
This way the user task will have that needed kernel task already running if needed (and kernel threads available).

4. At the end, that initial task (to be considered PID=0, being Parent of PID=1 and Parent ID of the kthreadadd task) will enter in an ininite idling loop:
```C
	cpu_startup_entry(CPUHP_ONLINE);
```
which is basically "do nothing":
```C
void cpu_startup_entry(enum cpuhp_state state)
{
	arch_cpu_idle_prepare();
	cpuhp_online_idle(state);
	while (1)
		do_idle();
}
```